python/ray/serve/_private/deployment_state.py

```diff
@@ -2112,9 +2199,9 @@ def update(
                 "Exception occurred trying to update deployment state:\n"
                 + traceback.format_exc()
             )
-            self._curr_status_info = DeploymentStatusInfo(
-                name=self.deployment_name,
+            self._curr_status_info.update(
                 status=DeploymentStatus.UNHEALTHY,
+                status_trigger=DeploymentStatusTrigger.UNSPECIFIED,
```

Comment:
We should distinguish b/w UNSPECIFIED (aka unset) vs INTERNAL_FAILURE (aka generic, unknown precisely)

Reply:
Should we expose users to a trigger named INTERNAL_FAILURE?

Reply:
Not sure fully understand angle of your question: 
 - We do expose INTERNAL_FAILURE as status in gRPC/HTTP handlers
 - We can name it differently as well, but 
 - We'd make sure we distinguish it from UNSPECIFIED

Reply:
I see, I'm OK with adding this as a trigger. I've named it `INTERNAL_ERROR` since `INTERNAL_FAILURE` is slightly inconsistent with triggers like `HEALTH_CHECK_FAILED`.